test/common/network/connection_impl_test.cc

```diff
@@ -1163,6 +1163,29 @@ TEST_P(ConnectionImplTest, DelayedCloseTimeoutNullStats) {
   callback();
 }
 
+class FakeReadFilter : public Network::ReadFilter {
+public:
+  FakeReadFilter() {}
+  ~FakeReadFilter(){
+    EXPECT_TRUE(callbacks_ != nullptr);
+    EXPECT_TRUE(callbacks_->connection().state() != Network::Connection::State::Open);
```

Comment:
When the destructor call ConnectionImpl socket_ members

Reply:
Do you mind adding an actual comment here about why this is here in addition to the one you have below?

/wait